samples/drivers/soc_flash_nrf/src/main.c

```diff
@@ -21,8 +25,31 @@
 #define FLASH_TEST_OFFSET2 0x41234
 #define FLASH_TEST_PAGE_IDX 37
 
+#if IS_ENABLED(CONFIG_BT_LL_NRFXLIB)
+static void bt_ready(int err)
+{
+	if (err) {
+		printf("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	printf("Bluetooth initialized\n");
+}
+#endif /* IS_ENABLED(CONFIG_BT_LL_NRFXLIB) */
+
 void main(void)
 {
+#if IS_ENABLED(CONFIG_BT_LL_NRFXLIB)
+    	int err;
+	/* We need to start the BLE Controller in order for the flash driver to
+	 * work. */
+	err = bt_enable(bt_ready);
```

Comment:
I don't think it is necessary to enable BLE. The BLE controller is initialized when the hci_driver is initialized. After that point of time you can use the flash APIs

Reply:
I assumed so myself, but for some reason this is not the case. Removing this code will cause the example to halt at the first write attempt. But, perhaps there is something else going on that's halting it.

Reply:
I think we need to debug why this happens. It is likely that something is wrong in the stack

Reply:
:+1: 

- [ ] debug why the flash gets stuck when not `bt_enable()` is called

Reply:
The signalling thread in the HCI driver was started in `hci_driver_open()` which is called by `bt_enable()`, I've moved the starting of the thread to initialization time in the other pull request.